src/utils/templates.ts

```diff
@@ -9,31 +9,45 @@ import * as https from 'https';
 import * as path from 'path';
 import { CopyDescriptor } from './copy-list';
 
-interface TemplateParam {
-  from: string[];
-  to: string;
-  exclude?: string[];
-}
+const templatesDir = path.resolve(__dirname, '../templates');
 
-export function readTemplates({ from, to, exclude = [] }: TemplateParam): CopyDescriptor[] {
-  const files = fs.readdirSync(path.resolve(...from), { withFileTypes: true });
-  const results: CopyDescriptor[] = [];
-  return files.reduce((prev, curr) => {
-    if (curr.isDirectory() && !exclude.includes(curr.name)) {
-      prev = prev.concat(readTemplates({ from: from.concat(curr.name), to, exclude }));
+function getTemplatePathsForDir(inputDir: string[], excludes: string[]): string[] {
+  const directoryEntries = fs.readdirSync(path.resolve(templatesDir, ...inputDir), { withFileTypes: true });
+  return directoryEntries.reduce((templates: string[], directoryEntry) => {
+    if (directoryEntry.isDirectory() && !excludes.includes(directoryEntry.name)) {
+      return [...templates, ...getTemplatePathsForDir([...inputDir, directoryEntry.name], excludes)];
     }
-    if (curr.isFile() && !exclude.includes(curr.name)) {
-      prev.push({
-        sourcePath: path.resolve(...from, curr.name),
-        targetFolder: path.resolve(to, ...from.slice(1)),
-        fileName: path.resolve(to, ...from.slice(1), path.basename(curr.name, '.mu'))
-      });
+    if (directoryEntry.isFile() && !excludes.includes(directoryEntry.name)) {
+      return [...templates, path.resolve(templatesDir, ...inputDir, directoryEntry.name)];
     }
-    return prev;
-  }, results);
+    return templates;
+  }, []);
+}
+
+export function getTemplatePaths(inputDirs: string[], excludes: string[] = []): string[] {
+  return inputDirs.reduce((templatePaths: string[], inputDir) => [...templatePaths, ...getTemplatePathsForDir([inputDir], excludes)], []);
```

Comment:
It's a matter of taste but I like this more:
```suggestion
  return inputDirs.reduce((templatePaths, inputDir) => [...templatePaths, ...getTemplatePathsForDir([inputDir], excludes)], [] as string[]);
```